tests/Backend/Mixed/SharingTest.php

```diff
@@ -48,6 +48,52 @@ public function testInvalidSharingType($sharingType, $isAsync): void
         }
     }
 
+    /** @dataProvider syncAsyncProvider */
+    public function testTryLinkBucketWithSameNameAsAlreadyCreatedBucketThrowUserException(bool $isAsync): void
```

Comment:
Ten test mi přijde dost divně postavený. Udělá custom bucket i když má bucket API-sharing a pak spoléhá na to, že je v cílovém projektu ten API-sharing vytvořený. Je to takové složité na přemýšlení. Klidně bych nasdílel ten API-sharing, když už je hotový. 

Nejlepší by bylo to napsat na čisto - tj. udělat si bucket v tom druhém projektu ručně, aby z toho testu bylo vidět, že tam je vytvořený a proto to failne. Tady to spoléhá na to, že víš, že se v každém projektu vytvoří API-sharing bucket. Ale tohle pak nebude fungovat paralelně, kvuli tomu jak funguje tady ten initTestBuckets, který zahodí všechno, co není API-sharing :/ Ale i tak by mi to přišlo lepší a čitelnější.

Ale hrozí tam vlastně ještě, že pak nebude fungovat ten assertCount, protože jich tam bude víc, když to nevyclearuješ tím "initTestBcukets".

Reply:
vychazi mi z toho co pises a i z toho co o tom vim, ze "lip to moc nejde". tak jsem to tam aspon zdokumentoval, aby ten test to o sobe rikal na prvni dobrou.